packages/publisher/src/main.ts

```diff
@@ -0,0 +1,93 @@
+import applicationinsights = require("applicationinsights");
+import * as yargs from "yargs";
+import { getSecret, Secret } from "./lib/secrets";
+import { Fetcher, loggerWithErrors, LoggerWithErrors } from "@definitelytyped/utils";
+import { currentTimeStamp } from "./util/util";
+import full from "./full";
+
+export default async function main(): Promise<void> {
+  const githubAccessToken = await getSecret(Secret.GITHUB_ACCESS_TOKEN);
+  const dry = !!(yargs.argv.dry || process.env.WEBHOOK_FORCE_DRY);
+
+  if (!githubAccessToken) {
+    console.log("The environment variable GITHUB_ACCESS_TOKEN must be set.");
+  } else {
+    console.log(`=== ${dry ? "DRY" : "PRODUCTION"} RUN ===`);
+    if (process.env.APPINSIGHTS_INSTRUMENTATIONKEY) {
+      applicationinsights.setup().start();
+      console.log("Done initialising App Insights");
+    }
+    const fetcher = new Fetcher();
+
+    try {
+      const fullOne = updateOneAtATime(async (log, timestamp) => {
+        log.info("");
+        log.info("");
+        log.info(`# ${timestamp}`);
+        log.info("");
+        log.info("Starting full...");
```

Comment:
one thing to test is whether the app service actually properly starts without a web request coming in. Not actually running a web server may confuse the App Service machinery.